pike/smb2.py

```diff
@@ -1829,19 +1829,23 @@ class LeaseRequest(CreateRequestContext):
     # This class handles V2 requests as well.  Set
     # the lease_flags field to a non-None value
     # to enable the extended fields
-    def __init__(self, parent):
+    def __init__(self, parent, do_v2_lease):
         CreateRequestContext.__init__(self, parent)
         self.lease_key = array.array("B", [0] * 16)
         self.lease_state = 0
         # V2 fields
-        self.lease_flags = None
-        self.parent_lease_key = None
-        self.epoch = None
+        self.lease_flags = LeaseFlags.SMB2_LEASE_FLAG_NONE
+        self.parent_lease_key = array.array("B", [0] * 16)
```

Comment:
`parent_lease_key` and `epoch` are only for v2 right but we set `isV2` later.

Reply:
maybe retain the old code an have new code under check `if do_v2_lease`, then set `isV2` within.